src/Sylius/Behat/Context/Api/Shop/OrderContext.php

```diff
@@ -61,13 +71,94 @@ public function iShouldBeAbleToAccessThisOrderDetails(): void
     }
 
     /**
-     * @Then /^my tax total should be ("[^"]+")$/
+     * @Then :promotionName should be applied to my order
      */
-    public function myTaxTotalShouldBe(int $taxTotal): void
+    public function shouldBeAppliedToMyOrder(string $promotionName): void
     {
-        $response = $this->client->show($this->sharedStorage->get('cart_token'));
+        Assert::notFalse(array_search($promotionName, array_column($this->getAdjustmentsForOrder(), 'label')));
```

Comment:
This extraction logic seems like a good candidate for placing it in separate private method